samcli/commands/local/lib/cfn_api_provider.py

```diff
@@ -103,3 +120,102 @@ def _extract_cloud_formation_stage(resources, stage_resource, collector):
 
         collector.stage_name = stage_name
         collector.stage_variables = stage_variables
+
+    def _extract_cloud_formation_method(self, resources, logical_id, api_resource, collector):
+        """
+        Extract APIs from AWS::ApiGateway::Method and work backwards up the tree to resolve and find the true path.
+
+        Parameters
+        ----------
+        logical_id : str
+            Logical ID of the resource
+
+        api_resource : dict
+            Resource definition, including its properties
+
+        resources: dict
+            All Resource definition, including its properties
+
+        stage_resource : dict
+            Stage Resource definition, including its properties
+
+        collector : ApiCollector
+            Instance of the API collector that where we will save the API information
+        """
+
+        properties = api_resource.get("Properties", {})
+        resource_id = properties.get("ResourceId")
+        rest_api_id = properties.get("RestApiId")
+        method = properties.get("HttpMethod")
+
+        resource_path = ""
+        if isinstance(resource_id, string_types):
+            resource = resources.get(resource_id)
+            if resource:
+                resource_path = self.resolve_resource_path(resources, resource, "")
+            else:
+                resource_path = resource_id  # In this case, the path was resolved to a string
+
+        integration = properties.get("Integration", {})
+        content_type = integration.get("ContentType")
+
+        content_handling = integration.get("ContentHandling")
+
+        if content_handling == CfnApiProvider.METHOD_BINARY_TYPE and content_type:
+            collector.add_binary_media_types(logical_id, [content_type])
+
+        routes = Route(methods=[method],
+                       function_name=self._get_integration_function_name(integration),
+                       path=resource_path)
+        collector.add_routes(rest_api_id, [routes])
+
+    def resolve_resource_path(self, resources, resource, current_path):
+        """
+        Extract path from the Resource object by going up the tree
+
+        Parameters
+        ----------
+        resources: dict
+            Dictionary containing all the resources to resolve
+
+        resource : dict
+            AWS::ApiGateway::Resource definition and its properties
+
+        current_path : str
+            Current path resolved so far
+        """
+
+        properties = resource.get("Properties", {})
+        parent_id = properties.get("ParentId")
+        path_part = properties.get("PathPart")
```

Comment:
At first I thought path_part was the separator (aka "/"), I know this matches what the API Gateway definition is but consider naming this to be `path` or `resource_path`. Maybe `path_part` does make sense and I am just the one jumping to conclusions.